docs/examples/snow-detection/snow-mode.py

```diff
@@ -0,0 +1,716 @@
+    ax.plot(data[v], alpha=0.2)
+ax.axhline(mppt_high_voltage, c='r', ls='--',
+           label='Maximum MPPT voltage: {} V'.format(mppt_high_voltage))
+ax.axhline(mppt_low_voltage, c='g', ls='--',
+           label='Minimum MPPT voltage: {} V'.format(mppt_low_voltage))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('Voltage [V]', fontsize='large')
+ax.legend(loc='lower left')
+plt.show()
+
+# %% Plot AC power relative to inverter nameplate limits
+
+fig, ax = plt.subplots(figsize=(10, 10))
+date_form = DateFormatter("%m/%d")
+ax.xaxis.set_major_formatter(date_form)
+for a in ac_power_cols:
+    ax.scatter(data.index, data[a], s=0.5, label=a)
+    ax.plot(data[a], alpha=0.2)
+ax.axhline(max_ac_power, c='r', ls='--',
+           label='Maximum allowed AC power: {} kW'.format(max_ac_power))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('AC Power [kW]', fontsize='large')
+ax.legend(loc='upper left')
+plt.show()
+
+# %% Filter data.
+# Identify periods where the system is operating off of its maximum power
+# point (MPP), and correct or mask. Conditions outside of the MPP cannot
+# be accurately modeled without external information on the system's
+# operating point. To allow us to make a valid comparison between system
+# measurements and modeled power at MPP, we set measurements collected below
+# the MPPT minimum voltage to zero, which emulates the condition where the
+# inverter turns off when it cannot meet the turn-on voltage.
+
+ac_power_cols_repeated = ac_power_cols + ac_power_cols + ac_power_cols
+for v, i, a in zip(dc_voltage_cols, dc_current_cols, ac_power_cols_repeated):
+
+    # Data where V < MPPT minimum
+    data.loc[data[v] < mppt_low_voltage, v] = 0
+    data.loc[data[v] < mppt_low_voltage, i] = 0
+    data.loc[data[v] < mppt_low_voltage, a] = 0
+
+    # Data where system is at Voc
+    data.loc[data[i] == 0, v] = 0
+
+    # Data where inverter is clipping based on AC power
+    mask1 = data[a] > max_ac_power
+    mask2 = clipping.geometric(ac_power=data[a], freq='15min')
+    mask3 = np.logical_or(mask1.values, mask2.values)
+
+    data.loc[mask3, v] = np.nan
+    data.loc[mask3, i] = np.nan
+    data.loc[mask3, a] = np.nan
+
+# %% Plot DC voltage for each combiner input with inverter nameplate limits
+
+fig, ax = plt.subplots(figsize=(10, 10))
+date_form = DateFormatter("%m/%d")
+ax.xaxis.set_major_formatter(date_form)
+for v in dc_voltage_cols:
+    ax.scatter(data.index, data[v], s=0.5, label=v)
+    ax.plot(data[v], alpha=0.2)
+ax.axhline(mppt_high_voltage, c='r', ls='--',
+           label='Maximum MPPT voltage: {} V'.format(mppt_high_voltage))
+ax.axhline(mppt_low_voltage, c='g', ls='--',
+           label='Minimum MPPT voltage: {} V'.format(mppt_low_voltage))
+ax.set_xlabel('Date', fontsize='large')
+ax.set_ylabel('Voltage [V]', fontsize='large')
+ax.legend(loc='lower left')
+plt.show()
+
+# %%
+
+# We want to exclude periods where array voltage is affected by horizon
```

Comment:
Remove this blank line for the text to be rendered as text instead of a code comment block